test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,305 @@
+	off = r_vector_index_ptr (offsets, 1);
+	mu_assert_eq_fmt (*off, UT64_MAX, "Unexpected offset", "%llu");
+	off = r_vector_index_ptr (offsets, 2);
+	mu_assert_eq_fmt (*off, UT64_MAX, "Unexpected offset", "%llu");
+	off = r_vector_index_ptr (offsets, 3);
+	mu_assert_eq_fmt (*off, (ut64)4440, "Unexpected offset", "%llu");
+	off = r_vector_index_ptr (offsets, 4);
+	mu_assert_eq_fmt (*off, (ut64)4447, "Unexpected offset", "%llu");
+	off = r_vector_index_ptr (offsets, 5);
+	mu_assert_eq_fmt (*off, UT64_MAX, "Unexpected offset", "%llu");
+
+	mu_end;
+}
+
+static bool test_r_core_annotated_code_print_json() {
+	RAnnotatedCode *code = get_hello_world ();
+	char *actual = malloc (900 * sizeof (char));
+	char *expected = "{\"code\":\"\\nvoid main(void)\\n{\\n    sym.imp.puts(\\\"Hello, World!\\\");\\n    return;\\n}\\n\",\"annotations\":[{\"start\":1,\"end\":5,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"datatype\"},{\"start\":6,\"end\":10,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"function_name\"},{\"start\":11,\"end\":15,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"keyword\"},{\"start\":23,\"end\":35,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"function_name\"},{\"start\":36,\"end\":51,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"constant_variable\"},{\"start\":23,\"end\":52,\"type\":\"offset\",\"offset\":4440},{\"start\":58,\"end\":64,\"type\":\"offset\",\"offset\":4447},{\"start\":58,\"end\":64,\"type\":\"syntax_highlight\",\"syntax_highlight\":\"keyword\"},{\"start\":58,\"end\":64,\"type\":\"offset\",\"offset\":4447}]}\n";
+	r_cons_new ();
+	r_cons_push ();
+	r_core_annotated_code_print_json (code);
+	strcpy (actual, r_cons_get_buffer ());
+	r_cons_pop ();
+	mu_assert_streq (actual, expected, "pdgj OUTPUT DOES NOT MATCH");
+	free (actual);
+	mu_end;
+}
+
+static bool test_r_core_annotated_code_print() {
+	RAnnotatedCode *code = get_hello_world ();
+	char *actual = malloc (900 * sizeof (char));
+	//Checking without line offset
+	char *expected_first = "\n"
+			       "void main(void)\n"
+			       "{\n"
+			       "    sym.imp.puts(\"Hello, World!\");\n"
+			       "    return;\n"
+			       "}\n";
+	r_cons_new ();
+	r_cons_push ();
+	r_core_annotated_code_print (code, NULL);
+	strcpy (actual, r_cons_get_buffer ());
+	r_cons_pop ();
+	mu_assert_streq (actual, expected_first, "pdg OUTPUT DOES NOT MATCH");
+	r_cons_pop ();
+
+	//Checking with offset - pdgo
+	RVector *offsets = r_annotated_code_line_offsets (code);
+	char *expected_second = "                  |\n"
+				"                  |void main(void)\n"
+				"                  |{\n"
+				"    0x00001158    |    sym.imp.puts(\"Hello, World!\");\n"
+				"    0x0000115f    |    return;\n"
+				"                  |}\n";
+	r_core_annotated_code_print (code, offsets);
+	strcpy (actual, r_cons_get_buffer ());
+	r_cons_pop ();
+	mu_assert_streq (actual, expected_second, "pdgo OUTPUT DOES NOT MATCH");
+	r_cons_pop ();
+	free (actual);
+	mu_end;
```

Comment:
Leak of code and offsets:
```suggestion
	r_vector_free (offsets);
	r_annotated_code_free (code);
	mu_end;
```